### ISSUE TITLE
Car Mechanic Simulator 2018 jittery mouse (645630)

### ISSUE THREAD

**Issue by Halpm** [ValveSoftware/Proton#1844]
# Compatibility Report
- Car Mechanic Simulator 2018
- 645630

## System Information
- GPU: GeForce GTX 960M
- Driver/LLVM version: 4.6.0 nvidia 390.77
- Kernel version:4.15.0-38-lowlatency
- Proton version: 3.16-3 Beta and 3.7.8 Beta
- Full system info: https://gist.github.com/Halpm/1fa5405076ed32170bc341acbcd6cac9

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

## Symptoms
When using Car mechanic simulator 2018 on ubuntu 18.04, the mouse cursor sometimes makes apparently unmotivated jumps to the top-left of the screen. This happens independently of changes in resolution, mouse sensitivity or whether or not the game is running in a window, and happens with both the track pad and usb mouse.

## Reproduction
Run game

**Comment by vanyasem** [comment]
A workaround for that issue is running a game in a virtual desktop (which is suggested by a Lutris installer). I can confirm that it works. https://gist.github.com/vanyasem/f787469ec8c50cdd3728340ef4c73269

P.s. Open Steam Overlay prior to alt-tabbing so that your mouse will get released properly

**Comment by vanyasem** [comment]
Also, the game seems to require some sort of video codes for in-game tutorial videos:
![20181031163442_1](https://user-images.githubusercontent.com/13350458/47791548-06126080-dd2b-11e8-8783-c3891a83ceec.jpg)

I guess you could watch them manually from `cms2018_Data/StreamingAssets/Movies/`, but that does sound like a huge pain to me

**Comment by vanyasem** [comment]
I tried to install `allcodecs` via winetricks, but that didn't solve the issue

Installing K-Lite Codec Pack Standard doesn't fix the issue as well

**Comment by EnderTurret** [comment]
The codec it's missing is in Windows Media Foundation. I managed to fix the codec issue by installing it using the script in [this](https://github.com/z0z0z/mf-install) repository.

**Comment by kisak-valve** [comment]
Hello @EnderTurret, the workaround you linked is legally problematic.

**Comment by EnderTurret** [comment]
Oh, I didn't know. Anyway, the codec it needs is in Windows Media Foundation.

**Comment by HonkingGoose** [comment]
Could any of you upload a Proton 4.2-3 log?
Extra brownie points if you give me information about what media files the game's trying to play. So things like media file type and codecs used.

**Comment by bbuske** [comment]
The codec issue has been fixed. There is a solution on prontondb about it. Just a small script. However, the mouse issue is still there. I am also running Ubuntu 18.4 LTS. Running the game in a virtual desktop does not work nor does the solution someone mentioned, regarding disabling mouse acceleration and lower sensitivity. It makes the issues a bit better, but the movement still is significant.

**Comment by Patola** [comment]
Yeah, I confirm, just bought the game and the mouse-jumping issue is there. It might seem small but it is annoying enough (someone said "suicide inducing") that it is a good reason for a refund. Otherwise with microsoft foundation platform libraries the game runs just fine.

**Comment by Patola** [comment]
Tried winetricks xinput, no dice. Tried many different venues including setting and unsetting virtual desktop, running in "EXCLUSIVE FULLSCREEN" (yeah, all caps) and normal modes, resetting the configuration, even tried changing to joystick mode but this way it just crashes to the desktop which is no good.

**Comment by thiras** [comment]
I confirm the issue still exist.
